Tests/SwiftSyntaxMacrosTest/MacroSystemTests.swift

```diff
@@ -145,6 +145,125 @@ public struct FileIDMacro: ExpressionMacro {
   }
 }
 
+extension PatternBindingSyntax {
+  /// When the variable is declaring a single binding, produce the name of
+  /// that binding.
+  fileprivate var singleBindingName: String? {
+    if let identifierPattern = pattern.as(IdentifierPatternSyntax.self) {
+      return identifierPattern.identifier.text
+    }
+
+    return nil
+  }
+}
+
+extension SyntaxProtocol {
+  /// Form a function name.
+  private func formFunctionName(
+    _ baseName: String, _ parameters: ParameterClauseSyntax?,
```

Comment:
`baseName` could just be a `TokenSyntax`. Then you could use `.init` instead of `"init"` below